File: easy/1260-shift-2d-grid/solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * 1260. Shift 2D Grid
 * Time: O(m * n)
 * Space: O(m * n)
 */
class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        int m = grid.length, n = grid[0].length;
        int total = m * n;
        k %= total; // full rotations are no-ops

        Integer[][] result = new Integer[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int newIdx = (i * n + j + k) % total;
                result[newIdx / n][newIdx % n] = grid[i][j];
            }
        }

        List<List<Integer>> out = new ArrayList<>();
        for (Integer[] row : result) {
            out.add(Arrays.asList(row));
        }
        return out;
    }
}

File: easy/1260-shift-2d-grid/solution.py
from typing import List


class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        """
        1260. Shift 2D Grid
        Time: O(m * n)
        Space: O(m * n)
        """
        m, n = len(grid), len(grid[0])
        total = m * n
        k %= total  # full rotations are no-ops

        result = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                new_idx = (i * n + j + k) % total
                result[new_idx // n][new_idx % n] = grid[i][j]
        return result
